src/test/java/org/codewars/utils/ReaderTest.java

```diff
@@ -14,8 +20,28 @@ public void testReadInt() {
     public void testReadBigInteger() {
     }
 
-    @Test
-    public void testReadDouble() {
+    @DataProvider(name = "readDoubleTestData")
+    private Object[][] readDoubleTestData() {
+        Object[][] testData = new Object[][] {
+                {new Scanner("7"), 7.0},
+                {new Scanner("0"), 0.0},
+                {new Scanner("-2,54"), -2.54},
+                {new Scanner(",666666660"), 0.66666666},
+                {new Scanner("abc"), Double.NaN},
+                {new Scanner(""), Double.NaN},
+                {new Scanner("18.5748"), Double.NaN}
+        };
+        return testData;
+    }
+    @Test(dataProvider = "readDoubleTestData")
+    public void testReadDouble(Scanner scanner, double expected) {
+        Reader reader = new Reader(scanner);
+        try {
+            double actual = reader.readDouble();
+            assertEquals(actual, expected);
+        } catch (NoSuchElementException e) {
+            assertTrue(Double.isNaN(expected));
+        }
```

Comment:
mb better would be to check on few inputs? I mean smth like `{new Scanner("sda\nadwqwdcsd sdqwqwwqe12qwdc xcsd\n20,1"), 20.1f}` (it works on readFloat).

Reply:
I've added a new input